packages/@react-stately/table/src/useTreeGridState.ts

```diff
@@ -70,21 +70,21 @@ export function useTreeGridState<T extends object>(props: TreeGridStateProps<T>)
 }
 
 // TODO: copied from useTreeState, perhaps make this accept multiple keys?
-function toggleKey<T>(set: 'all' | Set<Key>, key: Key, collection: ITableCollection<T>): Set<Key> {
-  let res;
-  if (set === 'all') {
-    res = new Set(collection.rows.filter(row => row.props.hasChildItems || row.props.children.length > collection.columnCount));
-    res.delete(key);
+function toggleKey<T>(currentExpandedKeys: 'all' | Set<Key>, key: Key, collection: ITableCollection<T>): Set<Key> {
+  let updatedExpandedKeys: Set<Key>;
+  if (currentExpandedKeys === 'all') {
+    updatedExpandedKeys = new Set(collection.rows.filter(row => row.props.childItems?.length > 0 || row.props.children.length > collection.columnCount).map(row => row.key));
```

Comment:
if a row has childItems that aren't loaded in yet and thus is an array of length 0 I think we will want to include it in the set of expanded keys maybe